File: crates/a3s/0.1.2/src/templates/python/skills/hello.py
# skills/hello.py — example a3s-code skill
from a3s_code import skill, ToolResult

@skill(
    name="hello",
    description="Say hello to someone",
    parameters={
        "name": {"type": "string", "description": "Name to greet"},
    },
)
async def hello(name: str) -> ToolResult:
    return ToolResult(content=f"Hello, {name}!")

File: crates/a3s/0.1.1/src/templates/typescript/skills/hello.ts
// skills/hello.ts — example a3s-code skill
import { skill, ToolResult } from "@a3s-lab/code";

export const hello = skill({
  name: "hello",
  description: "Say hello to someone",
  parameters: {
    name: { type: "string", description: "Name to greet" },
  },
  async execute({ name }: { name: string }): Promise<ToolResult> {
    return { content: `Hello, ${name}!` };
  },
});
